### DISCUSSION TITLE
Automation #5

### DISCUSSION THREAD

**Discussion by cleder** [Jesulayomy/SCHub#21] [Ideas]
Suggestions
- add `pre-commit` with `ruff` (`format` and `check`) to keep the python source clean
- add GitHub actions to run static and unit tests
-  add type annotations

**Comment by Jesulayomy** [comment]
@cleder Those sound great, I'll try to find some resources I could go through to learn how those work, you can recommend some if you know any. Haven't used ruff and the frontend was written without TS.

**Comment by cleder** [comment]
[A Tale of Two Kitchens - Hypermodernizing Your Python Code Base](https://dev.to/ldrscke/a-tale-of-two-kitchens-hypermodernizing-your-python-code-base-3jnh) for rationale and overview what can be done.

**Reply by Jesulayomy** [reply]
I've taken time to go through this and tested some use of ruff pre-commit. Will get an implementation rolling soon!
